### PR TITLE
feat: add token payload validation

### LINKED ISSUE #294: Process crashes when malformed identity token provided.
After calling an endpoint covered with LoginStrategy with malformed identity token like for instance
```json
{
    "identityToken": "sdfsdfsdf"
}
```
the whole nodejs process crashes after an exception is thrown in the following line: https://github.com/energywebfoundation/passport-did-auth/blob/1df78a155f12c09429d5652de3f61668850142c2/lib/LoginStrategy.ts#L146-L146

```
/mydid-protected-service/node_modules/passport-did-auth/dist/lib/LoginStrategy.js:110
            const iss = this.didUnification(payload.iss);
                                                    ^
TypeError: Cannot read properties of null (reading 'iss')
```

This makes it very easy to perform DOS attacks.

### PR DESCRIPTION
**Pull request by Harasz**

### CODE REVIEW

File: lib/LoginStrategy.ts [hunk 1]
```diff
@@ -156,11 +161,12 @@ export class LoginStrategy extends BaseStrategy {
     const userAddress = addressOf(userDid);
 
     try {
+      const claimBlockNumber =
```
Comment (JGiter) [inline lib/LoginStrategy.ts:164]: we can handle any type of claimData.blockNumber by
```
if (
   ...
latestBlock.sub(this.numberOfBlocksBack).ge(payload.claimData.blockNumber)
```

File: lib/LoginStrategy.ts [hunk 2]
```diff
@@ -334,4 +340,39 @@ export class LoginStrategy extends BaseStrategy {
     }
     return `${didParts[0]}:${didParts[1]}:${chainName}:${didParts[2]}`;
   }
+
+  validateTokenPayload(payload: unknown): payload is ITokenPayload {
```
Comment (JGiter) [inline lib/LoginStrategy.ts:344]:
```suggestion
  isTokenPayload(payload: unknown): payload is ITokenPayload {
```

Comment (Harasz) [reply lib/LoginStrategy.ts:344]: Could you explain why this is a better name?

Comment (JGiter) [reply lib/LoginStrategy.ts:344]: It is unclear what validation mean

**Review by JGiter — APPROVED** [review]

**Pull request merged**